src/main/java/jarvis/model/LessonNotes.java

```diff
@@ -1,24 +1,25 @@
 package jarvis.model;
 
 import java.util.Collection;
-import java.util.HashMap;
+import java.util.Comparator;
 import java.util.Map;
+import java.util.TreeMap;
 
 import jarvis.model.exceptions.StudentNotFoundException;
 
 /**
  * Represents the notes for a lesson in JARVIS.
  */
 public class LessonNotes {
-    private final StringBuilder overallNotes = new StringBuilder("Lesson Notes:\n");
-    private final HashMap<Student, StringBuilder> studentNotes;
+    private final StringBuilder generalNotes = new StringBuilder("General Notes:\n");
```

Comment:
I think `generalNotes` is better naming for what was `overallNotes` but maybe for now we could keep it as `overallNotes` first when you resolve the conflicts because unfortunately I already made many references to this using `overallNotes`. There would be quite some renaming to do for consistency, I can help to rename them next time